include/benchmark/benchmark.h

```diff
@@ -642,24 +643,25 @@ bool State::KeepRunningBatch(size_t n) {
     total_iterations_ -= n;
     return true;
   }
-  if (!started_ && !error_occurred_) {
+  if (!started_) {
     StartKeepRunning();
-    if (total_iterations_ >= n) {
-      total_iterations_-= n;
-    } else {
-      batch_leftover_  = n - total_iterations_;
-      total_iterations_ = 0;
+    if (!error_occurred_) {
+      if (total_iterations_ >= n) {
```

Comment:
how do we get here? on line 642 if total_iterations_ >= n, we return. if we're here then total_iterations_ must be < n.

i think you can avoid this check and improve the code generation.

Reply:
Good question. total_iterations_ is now set to 0 when State is initialized. StartKeepRunning() can set total_iterations_ to any positive number of iterations. For example, if max_iterations is 10 and KeepRunningBatch() is given a batch size of 100, we should hit this path. I see that I can fold it into the  if block just above the call to FinishKeepRunning(), though I wouldn't be surprised if it doesn't change generated code with optimizations enabled. Does this address your comment, or were you looking for something else?

On another front, how much does code generation matter this far down? The fast path should be hit every time within a benchmark except for up to two calls:
 1. The first call, which starts the benchmark
 2. The last call, either due to error or because it's the last iteration/batch.